packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1726,6 +1770,11 @@ export class ContainerRuntime
 		this.updateDocumentDirtyState(newState);
 	}
 
+	private async applyStashedIdAllocationOp(content: any) {
```

Comment:
Can you add a comment explaining what this is doing?
It looks like on every IdAllocation stashed op processing, it will recreate the idCompression from the stashedState in that op? If there are multiple IdAllocation ops that are stashed, isn't this expensive?
I have not looked into the Id compression's implementation but isn't it possible to store incremental state in the idAllocation ops?

Reply:
I added a comment clarifying what's going on here. The stashed ops implementation is admittedly a hack an should definitely be optimized. This API wasn't originally intended for stashed ops and is being "repurposed" to make these work. You're right that what's here is unnecessarily expensive. As implemented, we're stashing the compressor state in every op and deleting it once ops are ack'd. We should add some sort of incremental local state to ops so that we can re-hydrate from stashed ops without needing to store the entire state every time.